main.cpp

```diff
@@ -6319,673 +6330,698 @@ string cpu_reg(int val) {
                 }
             }
-        } else {
-            continue;
         }
-
-        if (strlen(rep) > 0) {
-            tuple<string,string,string> proc = {sm.front(), sm.back(), rep};
-            proc_insts.push_back(proc);
+    } else if (type == CDP) {
+        uint8_t opc1 = (val >> 20) & 0xf;
+        uint8_t CRn = (val >> 16) & 0xf;
+        uint8_t CRd = (val >> 12) & 0xf;
+        uint8_t coproc = (val >> 8) & 0xf;
+        uint8_t opc2 = (val >> 5) & 0x7;
+        uint8_t CRm = val & 0xf;
+
+        if (coproc == 0) {
+            // GPIO has no cdp instructions
+            return false;
+        } else if (coproc == 4|| coproc == 5) {
+            // DCP
+            bool ns = coproc == 5;
+            switch (opc1) {
+                case 0:
+                    if (CRm == 0) {
+                        snprintf(buf, buf_len, "dcp%s_init", ns ? "ns" : "");
+                    } else {
+                        snprintf(buf, buf_len, "dcp%s_add0", ns ? "ns" : "");
+                    }
+                break;
+                case 1:
+                    if (opc2 == 0) {
+                        snprintf(buf, buf_len, "dcp%s_add1", ns ? "ns" : "");
+                    } else {
+                        snprintf(buf, buf_len, "dcp%s_sub1", ns ? "ns" : "");
+                    }
+                break;
+                case 2:
+                    snprintf(buf, buf_len, "dcp%s_sqr0", ns ? "ns" : "");
+                break;
+                case 8:
+                    if (CRm == 2 && opc2 == 0) {
+                        snprintf(buf, buf_len, "dcp%s_norm", ns ? "ns" : "");
+                    } else if (CRm == 2 && opc2 == 1) {
+                        snprintf(buf, buf_len, "dcp%s_nrdf", ns ? "ns" : "");
+                    } else if (CRm == 0 && opc2 == 1) {
+                        snprintf(buf, buf_len, "dcp%s_nrdd", ns ? "ns" : "");
+                    } else if (CRm == 0 && opc2 == 2) {
+                        snprintf(buf, buf_len, "dcp%s_ntdc", ns ? "ns" : "");
+                    } else if (CRm == 0 && opc2 == 3) {
+                        snprintf(buf, buf_len, "dcp%s_nrdc", ns ? "ns" : "");
+                    } else {
+                        return false;
+                    }
+                break;
+                default:
+                    return false;
+            }
+        } else if (coproc == 7) {
+            // RCP
+            if (opc1 == 0 && CRd == 0 && CRn == 0 && CRm == 0) {
+                snprintf(buf, buf_len, "rcp_panic");
+            } else {
+                return false;
+            }
         }
+    } else {
+        return false;
     }
+    return true;
+}
 
-    auto out = get_file_idx(ios::out, 1);
+bool coprodis_command::execute(device_map &devices) {
+    auto in = get_file(ios::in);
+    std::stringstream buffer;
+    buffer << in->rdbuf();
 
-    if (proc_insts.size() == 0) {
-        *out << buffer.str();
-        return false;
-    }
+    std::regex instruction(
```

Comment:
Is this now unused? (and if so should it be deleted, to stop it being a red-herring?)